### PR TITLE
feat: improve `host` option

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [x] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
Yes

### Motivation / Use-Case

Improve `host` option by merging with `use-local-ip`. 

Now `host` option can accept - `local-ip`, `local-ipv4`, `local-ipv6`.

### Breaking Changes

`useLocalIp` option was removed.

### CODE REVIEW

File: lib/utils/createDomain.js [hunk 1]
```diff
@@ -9,8 +9,12 @@ function createDomain(options, server) {
...
   let hostname;
 
-  if (options.useLocalIp) {
-    hostname = ip.v4.sync() || '0.0.0.0';
+  if (options.host === 'local-ip') {
+    hostname = ip.v4.sync() || ip.v6.sync();
+  } else if (options.host === 'local-ipv4') {
+    hostname = ip.v4.sync();
+  } else if (options.host === 'local-ipv6') {
+    hostname = ip.v6.sync();
```
Comment (alexander-akait) [inline lib/utils/createDomain.js:17]: What about of `ipv6` is not available, what is the error? Maybe we need improve error message, need test, at least manually

Comment (alexander-akait) [reply lib/utils/createDomain.js:17]: Same for ipv4

Comment (snitin315) [reply lib/utils/createDomain.js:17]: It's like this, for now, WIP on improvements.

Screenshot 2021-03-26 at 7 44 31 PM

File: lib/Server.js [hunk 2]
```diff
@@ -519,6 +519,11 @@ class Server {
     }
 
     this.server.on('error', (err) => {
+      if (err.code === 'EADDRNOTAVAIL') {
+        this.logger.error(
+          `Address ${this.hostname} is not available. Try with a different value for "host" option`
+        );
+      }
       throw err;
```
[the remark this answers is not in the document]
Comment (alexander-akait) [reply lib/Server.js:527]: Let's improve:
-  this.logger.error(`You fancy message.\n${error}`), never eat the original message
- let's add test, as minimum for CLI

**Review by alexander-akait — COMMENTED** [review]
Need some improve

**Comment by alexander-akait** [comment]
Also need rebase

**Comment by alexander-akait** [comment]
Small improvements here, I was wrong about handling error, let's keep as is, because Node.js output the same message

File: lib/Server.js [hunk 3]
```diff
@@ -724,7 +724,16 @@ class Server {
   }
 
   listen(port, hostname, fn) {
-    this.hostname = hostname;
+    if (hostname === 'local-ip') {
+      this.hostname =
+        internalIp.v4.sync() || '0.0.0.0' || internalIp.v6.sync() || '::';
```
Comment (ylemkimon) [inline lib/Server.js:729]: Isn't this a dead code, since '0.0.0.0' can be never `false`?

Comment (alexander-akait) [reply lib/Server.js:729]: Oh, yep, we need change it to `internalIp.v4.sync() || internalIp.v6.sync() || '0.0.0.0'`

Comment (anshumanv) [reply lib/Server.js:729]: fixed here https://github.com/webpack/webpack-dev-server/pull/3128
